### ISSUE TITLE
Cloud Job Task spec MaxRetries should be a pointer

### ISSUE THREAD

**Issue by entigo-mart-erlenheim** [googleapis/google-api-go-client#2646]
I have a Cloud Job where I want MaxRetries to be 0, this Job is created through Cloud Deploy by using skaffold with rawYamls.
Problem is that MaxRetries field has omitempty json tag and thus the field is removed when setting it to 0. This causes the maxRetries to default to 3 instead. I'm using the v1 api structs if that makes any difference. The field is located in TaskSpec struct https://github.com/googleapis/google-api-go-client/blob/main/run/v1/run-gen.go#L5132

Potential solution would be to make the field into a pointer so 0 values could be marshaled and unmarshaled correctly.

**Comment by codyoss** [comment]
This is by design. To send fields that are default type values and omitted with omitempty please use ForceSendFields: https://pkg.go.dev/google.golang.org/api#hdr-ForceSendFields. Hope that helpes.

**Comment by entigo-mart-erlenheim** [comment]
> This is by design. To send fields that are default type values and omitted with omitempty please use ForceSendFields: https://pkg.go.dev/google.golang.org/api#hdr-ForceSendFields. Hope that helpes.

Thank you! My bad, I had completely missed the ForceSendFields field. Still getting used to the Google api differences when compared to AWS.

**Comment by codyoss** [comment]
No problem! Just an FYI we also offer a newer style of library that has some improved ergonomics. Here is the equivalent for the API you are using: https://pkg.go.dev/cloud.google.com/go/run@v1.3.7/apiv2

**Comment by entigo-mart-erlenheim** [comment]
Thank you for trying to help! I have now tried the suggested ForceSendFields and can report that it only helps with marshaling the Yaml, but the problem still persists because Cloud Deploy uses Skaffold to unmarshal the yaml by using Golang. This means that to Skaffold the field will still be 0 and count as empty field that won't be sent to the Cloud Run Job. MaxRetries ends up defaulting back to 3. I also created a [Cloud Forum support post ](https://www.googlecloudcommunity.com/gc/Developer-Tools/Setting-Cloud-Run-Job-MaxRetries-to-0-when-using-Deploy-with/td-p/767897) about this problem as I consider the API implementation to be flawed when it comes to yaml based workflows. Unless of course I'm missing something.
